src/attr.rs

```diff
@@ -195,15 +195,20 @@ impl<'s> Attributes<'s> {
         self.0.as_ref().map_or(true, |v| v.is_empty())
     }
 
+    #[must_use]
+    pub fn len(&self) -> usize {
+        self.0.as_ref().map_or(0, |v| v.len())
+    }
+
     /// Returns a reference to the value corresponding to the attribute key.
     #[must_use]
-    pub fn get(&self, key: &str) -> Option<&AttributeValue<'s>> {
+    pub fn get(&'s self, key: &str) -> Option<&'s AttributeValue<'s>> {
```

Comment:
Not according to the compiler - I think because the resulting iterator is parameterised by the lifetime, you need to annotate that it's the same lifetime as the originating struct.

Reply:
However, turns out you can use the old `.iter()` implementation and avoid those annotations. Up to you whether you'd prefer a more complicated `get` implementation with a slightly simpler signature (which may or may not make a difference to its use), or use the `into_iter` internally.

Reply:
I think it should be possible if you remove the <'s> from the AttributesIter?

Reply:
Yes, you're right - got too excited about sprinkling lifetime annotations everywhere.

Reply:
get doesn't need them but get_mut does, because iter() doesn't need them but iter_mut() does (see below)